lib/saml/url_service.rb

```diff
@@ -103,11 +120,15 @@ def build_sso_url(link_authn_context)
       new_url_settings.authn_context = link_authn_context
       saml_auth_request = OneLogin::RubySaml::Authrequest.new
 
-      saml_auth_request.create(new_url_settings, RelayState: relay_state_params)
+      query_params = { RelayState: relay_state_params }
+      query_params[:clientId] = client_id if client_id
```

Comment:
Since we don't get `client_id` when `saml_callback` is invoked, it will be missing when the user verifies during the login process.